### DISCUSSION TITLE
知识库更新中的各种问题

### DISCUSSION THREAD

**Discussion by applepieiris** [chatchat-space/Langchain-Chatchat#2025] [General]
**问题描述 / Problem Description**
将自己的文件放到content文件夹后，运行init knowledge_base重新创建了知识库（将原来知识库里的test.txt删除了已经）。web界面使用的时候模型依然会引用test.txt中的问题。

![1](https://github.com/chatchat-space/Langchain-Chatchat/assets/36785462/dc9ac624-7608-437b-a8dd-4ea075292624)

![2](https://github.com/chatchat-space/Langchain-Chatchat/assets/36785462/9418c2c6-043d-4998-8e35-374a28999d57)

**环境信息 / Environment Information**
- langchain-ChatGLM 版本/commit 号：commit b8aad55
- 是否使用 Docker 部署（是/否）：否
- 使用的模型（ChatGLM2-6B / Qwen-7B 等）：ChatGLM-6B / Model used (ChatGLM2-6B / Qwen-7B, etc.): ChatGLM2-6B
- 使用的 Embedding 模型（moka-ai/m3e-base 等）：moka-ai/m3e-base 
- 使用的向量库类型 (faiss / milvus / pg_vector 等)： faiss 
- 操作系统及版本 / Operating system and version: ubuntu
- Python 版本 / Python version: 3.8
- 其他相关环境信息 / Other relevant environment information:

**Comment by Alfer-Feng** [comment]
在知识库删除文档之后，有时候需重启web端服务才能更新知识库。但这不是一个长远之计，我上网查了查跟streamlit的缓存机制有关（显示删了，实际没删），这个需要资深人士解答。

**Comment by zRzRzRzRzRzRzR** [comment]
mark，我们查看一下

**Comment by zRzRzRzRzRzRzR** [comment]
检查一下新版本还有没有这个问题，就是删除之后的下一次回答还在搜索已经删除的知识库

**Reply by applepieiris** [reply]
谢谢，新版本没有这个问题了
